xfrd-tcp.c

```diff
@@ -573,12 +701,62 @@ xfrd_tcp_open(struct xfrd_tcp_set* set, struct xfrd_tcp_pipeline* tp,
 	tp->tcp_r->fd = fd;
 	tp->tcp_w->fd = fd;
 
+	/* Check if an tls_auth name is configured which means we should try to
+	   establish an SSL connection */
+	if (zone->master->tls_auth_options &&
+		zone->master->tls_auth_options->auth_domain_name) {
+#ifdef HAVE_TLS_1_3
+		if (!setup_ssl(tp, set, zone->master->tls_auth_options->auth_domain_name)) {
+			log_msg(LOG_ERR, "xfrd: Cannot setup TLS on pipeline for %s to %s",
+					zone->apex_str, zone->master->ip_address_spec);
+			close(fd);
+			xfrd_set_refresh_now(zone);
+			return 0;
+		}
+		tp->handshake_done = 0;
+		if(!ssl_handshake(tp)) {
+			if(tp->handshake_want == SSL_ERROR_SYSCALL) {
+				log_msg(LOG_ERR, "xfrd: TLS handshake failed "
+					"for %s to %s: %s", zone->apex_str,
+					zone->master->ip_address_spec,
+					strerror(errno));
+
+			} else if(tp->handshake_want == SSL_ERROR_SSL) {
+				if(SSL_get_verify_result(tp->ssl) == X509_V_OK)
+					log_crypto_err("xfrd: TLS handshake failed");
```

Comment:
```suggestion
			} else if(tp->handshake_want == SSL_ERROR_SSL) {
				log_crypto_err("xfrd: TLS handshake failed");
```
The get_verify_result is only necessary when the handshake has completed successfully? With SSL_ERROR_SSL we know it is wrong and the log_crypto_err function prints the details that are in the error stack.

Reply:
The verify function itself also prints an error message if the name doesn't match.
I though to be smart and preventing double printing of the message by first testing if it is something else than a verification problem (since `SSL_get_verify_result()` results `X509_V_OK` if no verification has taken place yet.
But I noticed with the xot tdir test that no error is logged when the CA doesn't match. So I guess it is better to have double error reporting than none.
Will remove the extra `SSL_get_verify_result(tp->ssl) == X509_V_OK` test

Reply:
Ah, it was actually a type in the xot test that caused the missing CA not to be printed, but I think it is fine (and safe) to just report the error twise (once during verification, and once when the handshake fails). The fixes are in commits 78eace0 and 9603146